File: azor-the-chatdog-js/src/commands/sessionDisplay.ts
/**
 * Session display command (matches Python version)
 */

import type { ChatSession } from '../session/chatSession.js';
import { printInfo, printUser, printAssistant } from '../cli/console.js';

/**
 * Display the full conversation history of current session
 */
export function displaySessionHistory(session: ChatSession): void {
  const history = session.getHistory();

  if (history.length === 0) {
    printInfo('Session history is empty.');
    return;
  }

  printInfo(
    `\n--- FULL SESSION HISTORY (${session.id}, ${history.length} message(s)) ---`
  );

  for (let i = 0; i < history.length; i++) {
    const msg = history[i];
    const role = msg.role;
    const displayRole = role === 'user' ? 'YOU' : session.assistantName;
    const text = msg.parts[0]?.text || '';

    // Display with appropriate function based on role
    if (role === 'user') {
      printUser(`\n[${i + 1}] ${displayRole}:`);
      printUser(text);
    } else {
      printAssistant(`\n[${i + 1}] ${displayRole}:`);
      printAssistant(text);
    }
  }

  printInfo('--------------------------------------------------------');
}

File: azor-the-chatdog-py/src/commands/session_display.py
from typing import List, Dict
from cli import console

def display_full_session(history: List[Dict], session_id: str, assistant_name: str):
    """
    Displays the full session history.
    
    Args:
        history: List of dicts: {"role": "user|model", "parts": [{"text": "..."}]}
        session_id: Session ID
        assistant_name: Assistant name to display
    """
    if not history:
        console.print_info("Session history is empty.")
        return

    console.print_info(f"\n--- FULL SESSION HISTORY ({session_id}, {len(history)} message(s)) ---")
    
    for i, content in enumerate(history):
        # Handle universal dictionary format
        role = content.get('role', '')
        display_role = "YOU" if role == "user" else assistant_name
        
        # Extract text from parts
        text = ""
        if 'parts' in content and content['parts']:
            text = content['parts'][0].get('text', '')
        
        # Display with appropriate function
        if role == "user":
            console.print_user(f"\n[{i+1}] {display_role}:")
            console.print_user(f"{text}")
        else:
            console.print_assistant(f"\n[{i+1}] {display_role}:")
            console.print_assistant(f"{text}")
            
    console.print_info("--------------------------------------------------------")
